yt_dlp/extractor/instagram.py

```diff
@@ -404,11 +404,12 @@ def _real_extract(self, url):
             r'window\.__additionalDataLoaded\s*\(\s*[^,]+,\s*', webpage, 'additional data', video_id, fatal=False)
         product_item = traverse_obj(additional_data, ('items', 0), expected_type=dict)
         if product_item:
-            product_item.update(general_info)
-            return self._extract_product(product_item)
-        media = traverse_obj(additional_data, ('graphql', 'shortcode_media'), 'shortcode_media', expected_type=dict) or {}
+            general_info.update(product_item)
+            return self._extract_product(general_info)
 
-        media.update(general_info)
+        media = traverse_obj(additional_data, ('graphql', 'shortcode_media'), 'shortcode_media', expected_type=dict) or {}
+        general_info.update(media)
+        media = general_info
```

Comment:
why? Just don't use `media` below

Reply:
Previously the general_info overwrites any same key value of media. But general_info can have some empty key values which gets overwritten on media. Since media is second in extraction priority it's what that overwrites general_info

Reply:
10743d47d108d358c845b8e065682a9e963b64a1 is what I mean

Reply:
yea it makes more sense